File: llm_written/gpt-5/bfs/cpp/bfs_seq.cpp
#include "bfs_seq.hpp"
#include "graph.h"
#include <queue>
#include <unordered_map>
#include <vector>

std::vector<int> bfs_seq(Graph& g, int start_vertex) {
    std::vector<int> result;

    if (g.vertices.count(start_vertex) == 0) {
        return result;
    }

    std::unordered_map<int, bool> visited;
    std::queue<int> q;

    visited[start_vertex] = true;
    q.push(start_vertex);

    while (!q.empty()) {
        int current = q.front();
        q.pop();
        result.push_back(current);

        for (int neighbor : g.vertices.at(current)) {
            if (!visited[neighbor]) {
                visited[neighbor] = true;
                q.push(neighbor);
            }
        }
    }

    return result;
}

File: baselines/bfs/go/bfs_seq.go
package bfsgo

type Graph struct {
	Vertices map[int][]int
}

func (g *Graph) AddEdge(from, to int){
	if g.Vertices == nil {
		g.Vertices = make(map[int][]int)
	}
	g.Vertices[from] = append(g.Vertices[from], to)
	g.Vertices[to] = append(g.Vertices[to], from)
}


func Bfs(g Graph, startVertex int) []int {
	if _, exists := g.Vertices[startVertex]; !exists {
		return []int{}
	}

	visited := map[int]bool{startVertex: true}
	result := []int{}
	queue := []int{startVertex}

	for len(queue) > 0 {
		current := queue[0]
		queue = queue[1:]
		result = append(result, current)

		for _, neighbor := range g.Vertices[current] {
			if !visited[neighbor] {
				visited[neighbor] = true
				queue = append(queue, neighbor)
			}
		}
	}

	return result
}